File: Backends/C#/SuperHeroApi/SuperHeroDomain/Infrastructure/Query/PagedServiceResponseBase.cs
using SuperHeroCore;
using System;
using System.Collections.Generic;
using System.Text;

namespace SuperHeroDomain.Infrastructure.Query
{
    public class PagedServiceResponseBase : ServiceResponse, IPagedResponse
    {
        public int? CurrentPage { get; set; }

        public int? PageCount { get; set; }

        public int? PageSize { get; set; }

        public int? TotalItems { get; set; }

        public static T FromIPagedResponse<T>(IPagedResponse paged)
            where T : PagedServiceResponseBase, new()
        {
            return new T
            {
                CurrentPage = paged.CurrentPage,
                PageCount = paged.PageCount,
                PageSize = paged.PageSize,
                TotalItems = paged.TotalItems
            };
        }
    }
}

File: Backends/TypeScript/src/SuperHeroDomain/Infrastructure/PagedServiceResponseBase.ts
import { ServiceResponse } from "../../SuperHeroCore/ServiceResponse";
import { IPagedResponse } from "./Query/PagedResponse.interface";

export class PagedServiceResponseBase extends ServiceResponse implements IPagedResponse{
    current_page: number = 0;
    page_count: number = 0;
    page_size: number = 0;
    total_items: number = 0;
}
    